packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -642,6 +642,8 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 							this.retryOf,
 						);
 					}
+				} finally {
+					eventEmitter.emit('saveWorkflowStatistics', this.workflowData, fullRunData);
```

Comment:
I'd suggest to introduce an event type enum instead of using magic strings. Also, events should be emitted when some relevant thing happened and denote that clearly in their name, e.g. "workflowExecutionCompleted" (as opposed to an instruction/command like "saveWorkflowStatistics" which would describe what (one of potentially many) event _listeners_ does.)